### PR TITLE
Document behaviour when setting both on_step=True and on_epoch=True in self.log

### LINKED ISSUE #4216: Checkpoint is saving the model based on the last val_metric_step value and not val_metric_epoch
## 🐛 Bug

Checkpoint callback did not save some models even thought they achieved better result in the monitored metric, than the currently top k saved models

### Expected behavior

Checkpoint callback saving the best scoring models based on a metric

### Environment

I am using pytorch-lightning 1.0.2

#### Update:
I changed the checkpoint call back to add the value I am monitoring to the name of the saved checkpoint, what I notice it's not the epoch value, but the last step in the epoch value, so it's not taking the metric average value, but taking only the last one.

### PR DESCRIPTION
**Pull request by awaelchli**
Just a minor documentation update for self.log.

Fixes #4216

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

File: docs/source/logging.rst [hunk 1]
```diff
@@ -85,7 +85,13 @@ The :func:`~~pytorch_lightning.core.lightning.LightningModule.log` method has a
 * `logger`: Logs to the logger like Tensorboard, or any other custom logger passed to the :class:`~pytorch_lightning.trainer.trainer.Trainer`.
 
 
-.. note:: Setting `on_epoch=True` will accumulate your logged values over the full training epoch.
+.. note::
+
+    -   Setting ``on_epoch=True`` will accumulate your logged values over the full training epoch.
```
Comment (s-rog) [inline docs/source/logging.rst:90]: "accumulate" says `sum` to me, but I thought the default behavior is `mean`? Perhaps "reduce" is more appropriate?

File: docs/source/logging.rst [hunk 2]
```diff
@@ -85,7 +85,13 @@ The :func:`~~pytorch_lightning.core.lightning.LightningModule.log` method has a
...
+
+    -   Setting both ``on_step=True`` and ``on_epoch=True`` will create two keys per metric you log with
+        suffix ``_step`` and ``_epoch``, respectively. You can refer to these keys e.g. in the `monitor`
+        argument of :class:`~pytorch_lightning.callbacks.model_checkpoint.ModelCheckpoint` or in the graphs plotted to TensorBoard.
```
Comment (SkafteNicki) [inline docs/source/logging.rst:94]:
```suggestion
        argument of :class:`~pytorch_lightning.callbacks.model_checkpoint.ModelCheckpoint` or in the graphs plotted to the logger of your choice.
```

File: docs/source/logging.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source/logging.rst:90]: both are equally valid imo.

Comment (awaelchli) [reply docs/source/logging.rst:90]: the docs in this section do not mention the default reduction. 
We should say that the default is torch.mean

Comment (tchaton) [inline docs/source/logging.rst:90]: - Setting ``on_epoch=True`` will cache all your logged values during the full training epoch and perform a reduction ``on_epoch_end`. We recommend using ``Metric API`` when working with custom reduction.

**Review by tchaton — CHANGES_REQUESTED** [review]
Would you mind adding a reference to Metric API ?

**Review by ydcjeff — APPROVED** [review]

**Review by williamFalcon — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: docs/source/logging.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source/logging.rst]: @tchaton sry for the delay, I added your suggestion and linked metrics

Comment (awaelchli) [reply docs/source/logging.rst]: If you are happy with my changes, could you re-review so we can merge?

**Comment by awaelchli** [comment]
@tchaton are you ok with the changes now?

**Review by s-rog — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Comment by awaelchli** [comment]
@s-rog there is still one "request changes" review from @tchaton, so until this is gone it can't be merged :) So updating the branch will not help unfortunetely :)

**Comment by s-rog** [comment]
Ah you're right, missed that

**Review by tchaton — APPROVED** [review]
LGTM !
